### PR TITLE
Fix lang helpers

### PR DESCRIPTION
**Pull request by denismakogon**
Closes: https://github.com/fnproject/fn/issues/506
Closes: #89

### CODE REVIEW

**Comment by Lawouach** [comment]
Just tested locally and it fixed the error I had reported. Thank you for being so swift about it :)

**Comment by zootalures** [comment]
python2 seems to fail: 
```
$ fn init --runtime python2.7

$ fn build 
Building image playgronud:0.0.1
Sending build context to Docker daemon   5.12kB
Step 1/6 : FROM fnproject/python:2.7.13
2.7.13: Pulling from fnproject/python
88286f41530e: Already exists
ee67ab14fe7c: Pull complete
19d8fd8d5a29: Pull complete
Digest: sha256:54b06b8ddc46ee58bc2fd714bce379b7819d97e43b57cffdd3bb76aab9d09f73
Status: Downloaded newer image for fnproject/python:2.7.13
 ---> 803eb841a1ac
Step 2/6 : WORKDIR /function
Removing intermediate container 812ca5cc5076
 ---> 59c46fc5a3a5
Step 3/6 : ADD requirements.txt /function/
 ---> f22ae9b58bdb
Step 4/6 : RUN pip2 install -r requirements.txt
 ---> Running in aa3c0bfd5d6d
/bin/sh: pip2: not found
The command '/bin/sh -c pip2 install -r requirements.txt' returned a non-zero code: 127
ERROR: error running docker build: exit status 127
```

**Comment by Lawouach** [comment]
I wonder if for python2, it's not just `pip`.

**Comment by zootalures** [comment]
Python 3 also not working as it's trying to exec `python2`

Python 3 not working 

```
/t/playgronud ❯❯❯ fn init --runtime python3.6

        ______
       / ____/___
      / /_  / __ \
     / __/ / / / /
    /_/   /_/ /_/

Runtime: python3.6
func.yaml created.
/t/playgronud ❯❯❯ l
func.yaml
/t/playgronud ❯❯❯ cat > func.py
import sys
import os
import json

sys.stderr.write("Starting Python Function\n")

name = "World"

try:
  if not os.isatty(sys.stdin.fileno()):
    try:
      obj = json.loads(sys.stdin.read())
      if obj["name"] != "":
        name = obj["name"]
    except ValueError:
      # ignore it
      sys.stderr.write("no input, but that's ok\n")
except:
  pass

print "Hello", name, "!"
/t/playgronud ❯❯❯ ls
func.py   func.yaml
/t/playgronud ❯❯❯ fn run
Building image playgronud:0.0.1
Sending build context to Docker daemon  4.096kB
Step 1/4 : FROM fnproject/python:3.6
 ---> bb31b912c785
Step 2/4 : WORKDIR /function
 ---> Using cache
 ---> 0ae6b36ed462
Step 3/4 : ADD . /function/
 ---> 80ef1b9e4c83
Step 4/4 : ENTRYPOINT ["python2", "func.py"]
 ---> Running in 64465a1c246f
Removing intermediate container 64465a1c246f
 ---> 031092156969
Successfully built 031092156969
Successfully tagged playgronud:0.0.1
docker: Error response from daemon: OCI runtime create failed: container_linux.go:295: starting container process caused "exec: \"python2\": executable file not found in $PATH": unknown.
ERROR: exit status 127
```



Autodetect broken (as 'python' is gone) :
```
(with func.py) 

$fn init

        ______
       / ____/___
      / /_  / __ \
     / __/ / / / /
    /_/   /_/ /_/

Found python function, assuming python runtime.
Init does not support the python runtime, you'll have to create your own Dockerfile for this functionERROR: could not detect entrypoint or cmd for python, use --entrypoint and/or --cmd to set them explicitly
```

**Comment by Lawouach** [comment]
I realised I said it worked because I tried my limited use case, I apologise if I created some noise here.

File: langs/base.go [hunk 1]
```diff
@@ -22,8 +22,10 @@ func GetLangHelper(lang string) LangHelper {
 		return &NodeLangHelper{}
 	case "ruby":
 		return &RubyLangHelper{}
-	case "python":
-		return &PythonLangHelper{}
+	case "python2.7":
```
[the remark this answers is not in the document]
Comment (denismakogon) [reply langs/base.go]: addressed

**Comment by denismakogon** [comment]
@zootalures @Lawouach all comments were addressed

**Comment by zootalures** [comment]
python 3 still not working with fn init: 

```
/t/playgronud ❯❯❯  fn init --runtime python3.6

        ______
       / ____/___
      / /_  / __ \
     / __/ / / / /
    /_/   /_/ /_/

Runtime: python3.6
func.yaml created.

/t/playgronud ❯❯❯  docker pull fnproject/python:3.6
3.6: Pulling from fnproject/python
b56ae66c2937: Already exists
7febe839de39: Pull complete
e87b6dbf5eb8: Pull complete
Digest: sha256:268a9072e37cc1935d93c88393ccf8755047429eb26d188ca8de8728ec30af00
Status: Downloaded newer image for fnproject/python:3.6

/t/playgronud ❯❯❯ fn run
Building image playgronud:0.0.1
Sending build context to Docker daemon  4.096kB
Step 1/4 : FROM fnproject/python:3.6
 ---> f5f2a0e3267f
Step 2/4 : WORKDIR /function
Removing intermediate container 31f306551d5e
 ---> 22fa98d17783
Step 3/4 : ADD . /function/
 ---> d5055632743c
Step 4/4 : ENTRYPOINT ["python", "func.py"]
 ---> Running in 7e47d39a5d51
Removing intermediate container 7e47d39a5d51
 ---> accfec93ab6f
Successfully built accfec93ab6f
Successfully tagged playgronud:0.0.1
docker: Error response from daemon: OCI runtime create failed: container_linux.go:295: starting container process caused "exec: \"python\": executable file not found in $PATH": unknown.
ERROR: exit status 127
````

**Comment by denismakogon** [comment]
Okay, this is really annoying. Had false expectation to have aliases in fnproject/python images =/
Moving to versioned executables.
Manually tested.
```
 fn run
Building image fnproject:0.0.1
Sending build context to Docker daemon  669.6MB
Step 1/4 : FROM fnproject/python:3.6
3.6: Pulling from fnproject/python
b56ae66c2937: Pull complete 
7febe839de39: Pull complete 
e87b6dbf5eb8: Pull complete 
Digest: sha256:268a9072e37cc1935d93c88393ccf8755047429eb26d188ca8de8728ec30af00
Status: Downloaded newer image for fnproject/python:3.6
 ---> f5f2a0e3267f
Step 2/4 : WORKDIR /function
 ---> 15c81e31f57e
Removing intermediate container b0f1cbfa676f
Step 3/4 : ADD . /function/
 ---> 9da09cc9a470
Step 4/4 : ENTRYPOINT python3 func.py
 ---> Running in a169a21bd5dd
 ---> 1f3a2faf2ec0
Removing intermediate container a169a21bd5dd
Successfully built 1f3a2faf2ec0
Successfully tagged fnproject:0.0.1
```

**Comment by zootalures** [comment]
manually tested python, python 2.7 python 3.6 autodetect 

php is still broken : 
```
/t/playground ❯❯❯ fn init --runtime php
Runtime: php
func.yaml created.
/t/playground ❯❯❯ ls
func.yaml
func.php
/t/playground ❯❯❯ fn run
Building image playground:0.0.1
Sending build context to Docker daemon  4.096kB
Error response from daemon: Dockerfile parse error line 3: FROM requires either one or three arguments
ERROR: error running docker build: exit status 1
/t/playground ❯❯❯ less func.yaml                                                                                                                                                                                                                                                                                                                                                                                                         ⏎
version: 0.0.1
runtime: php
entrypoint: php func.php
```

**Comment by denismakogon** [comment]
@zootalures PHP is out of scope for my case. Don't know how to fix that. AFAIK someone already working on that, at least have seen in Slack.

**Pull request merged**